tests/test_python_support_policy.py

```
@@ -0,0 +1,199 @@
+def _matrix_python_versions(job: Mapping[str, object]) -> tuple[tuple[int, int], ...]:
+    strategy = job.get("strategy")
+    assert isinstance(strategy, dict)
+    matrix = strategy.get("matrix")
+    assert isinstance(matrix, dict)
+    raw_versions = matrix.get("python-version")
+    assert isinstance(raw_versions, list)
+
+    versions: list[tuple[int, int]] = []
+    for raw_version in raw_versions:
+        major, separator, minor = str(raw_version).partition(".")
+        assert separator and major.isdigit() and minor.isdigit(), raw_version
```

Ruff (0.15.21)

[warning] Assertion should be broken down into multiple parts

(PT018)